packages/shared/lib/core/network/utils/buildPersistedNetworkFromNodeInfoResponse.ts

```diff
@@ -10,14 +8,14 @@ export function buildPersistedNetworkFromNodeInfoResponse(
 ): IPersistedNetwork {
     const networkName = nodeInfoResponse?.nodeInfo?.protocol.networkName
     const networkId = getNetworkIdFromNetworkName(networkName)
-    const name = networkId === NetworkId.Custom ? networkName : DEFAULT_NETWORK_METADATA[networkId]?.name
+    const bech32Hrp = nodeInfoResponse?.nodeInfo?.protocol.bech32Hrp ?? DEFAULT_NETWORK_METADATA?.[networkId]?.bech32Hrp
+    const name = networkName ?? DEFAULT_NETWORK_METADATA?.[networkId]?.name
     const _coinType = coinType ?? COIN_TYPE[networkId] ?? 1
     return {
         id: networkId,
         name: name ?? 'Unknown Network',
         coinType: _coinType,
-        protocol: nodeInfoResponse?.nodeInfo?.protocol,
-        baseToken: { standard: TokenStandard.BaseToken, ...nodeInfoResponse?.nodeInfo?.baseToken },
+        bech32Hrp: bech32Hrp ?? 'Unknown Network',
```

Comment:
Why did you add `'Unknown Network'` as fallback value here now? This will likely break it as well and it doesn't make sense to use a human-readable phrase for this either